### DISCUSSION TITLE
Error with polars tutorial : `sep` should be `separator` in polars-tuto.ipynb

### DISCUSSION THREAD

**Discussion by lesteve** [InseeFrLab/onyxia#788] [Q&A]
Hi there,

I was curious to give a go to Onyxia following the [Reddit post](https://old.reddit.com/r/france/comments/1bune1b/le_service_public_d%C3%A9veloppe_un_logiciel_open/).

I clicked around and went to the polars tutorials which has some issue probably because polars has changed some things since this notebook was created? Not sure where it is best to give some feed-back to I thought Discussions may be a good starting point?

Here is a screenshot in case it helps about the tutorial I am trying to follow:
![image](https://github.com/InseeFrLab/onyxia/assets/1680079/161bc190-dd23-4207-b120-f349145a5ddc)

I am just curious, is there a way to see the notebook where the content comes from, is it
https://github.com/InseeFrLab/ssphub/blob/main/post/polars/polars-tuto.ipynb?

Is there a way to contribute fixes to this notebook if I get the time? No promises at all, I am mostly curious about Onyxia.

**Comment by garronej** [comment]
Hello @lesteve,  
Thanks for your interest in the project.
I have made a video to try answer your questions:  

https://youtu.be/4E_9D-4Qlqg

**Comment by linogaliana** [comment]
Hi @lesteve ! Thanks a lot for your feedback ! Discussions here are a great place to chat about tutorials, we welcome any feedback that helps us improve them.

Sorry about the outdated notebook, indeed it has not been updated for a while and since I am not using that often polars I forgot to update it. I managed to identify and correct the errors. The notebook should now work as expected 😃 

I would be happy to see any suggestions you might do for improving this notebook. 

It is indeed stored in https://github.com/InseeFrLab/ssphub/blob/main/post/polars/polars-tuto.ipynb . This notebook has been generated using `Quarto Markdown`, the source code is https://github.com/InseeFrLab/ssphub/blob/main/post/polars/polars-tuto.qmd . In SSPCloud instance, using a vscode service, I generated the file from terminal

```
quarto polars-tuto.qmd --to ipynb
```

That being said, we could directly create the tutorial from a Jupyter Notebook. I was initially more confortable to generate the notebook from Quarto because the latter works better with version control than Jupyter but I can adapt to your favored approach.

**Comment by lesteve** [comment]
Nice, thanks a lot for your answers and fixing the notebooks! There was also a deprecation `groupby` -> `group_by` I think somewhere IIRC which seems to match the [polars doc](https://docs.pola.rs/py-polars/html/reference/dataframe/api/polars.DataFrame.groupby.html)

The quarto approach + generated notebook makes sense. FYI we did something similar for our scikit-learn MOOC where notebooks are generated from `.py` files with Jupytext and you can run `.py` with VSCode cell-by-cell which we found very useful while developing the material and for obvious version control reasons. More details at: https://github.com/inria/scikit-learn-mooc
